test/extended/machines/scale.go

```diff
@@ -186,6 +187,35 @@ func scaleMachineSet(name string, replicas int) error {
 }
 
 var _ = g.Describe("[sig-cluster-lifecycle][Feature:Machines][Serial] Managed cluster should", func() {
+
+	var (
+		c      *kubernetes.Clientset
+		dc     dynamic.Interface
+		helper *bmhelper.BaremetalTestHelper
+	)
+
+	g.BeforeEach(func() {
+
+		cfg, err := e2e.LoadConfig()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		c, err = e2e.LoadClientset()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		dc, err = dynamic.NewForConfig(cfg)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		// For baremetal platforms, an extra worker must be previously
+		// deployed to allow subsequent scaling operations
+		helper = bmhelper.NewBaremetalTestHelper(dc)
```

Comment:
I think it would be better to do these operations only if platformType is equal to Baremetal. I know `NewBaremetalTestHelper` is intelligent enough to skip for other platform types but from this test's perspective, test just can not rely on it.

Reply:
Considering that we are slightly enlarging the scopes of baremetal related operators (ie CBO) to other platforms, my intention was to have a more granular helper class. If we could have the same support on other platforms in future, this test could continue to work. Also, the setup/teardown code would be less cumbersome (across different tests), given that the helper will not required it (personally, I'd limit the platform type check only to decide if a whole test must be skipped or not).
I'm not very happy of the teardown method name, probably naming it `Teardown` would make the helper class even more generic.